File: 2015/05/part2.js
const fs = require('fs');
const input = fs.readFileSync('input.txt', 'utf-8').split('\n');

let counter = 0;
for (let str of input) {
	if (str.match(/(\w{2}).*\1/g) && str.match(/(\w)\w\1/g))
		counter++;
}
console.log(counter)

File: 2019/03/part2.py
with open('input.txt', 'r') as f:
	data = f.read().split('\n')

# i wish i thought of this from the start
grid = {}
crossings = []
lineNo = 0
for line in data:
	steps = 0
	pos = [0, 0]
	line = line.split(',')
	for segment in line:
		dirChar = segment[0]
		dirV = [0, 1] if dirChar == "U" else [0, -1] if dirChar == "D" else [1, 0] if dirChar == "R" else [-1, 0]
		for i in range(1, int(segment[1:])+1):
			pos[0] += dirV[0]
			pos[1] += dirV[1]
			steps += 1
			if lineNo == 0:
				grid[str(pos)] = steps
				continue
			try:
				crossings.append(grid[str(pos)] + steps)
				print(pos, grid[str(pos)], steps)
			except:
				continue
	lineNo += 1

print(min(crossings))